src/types/index.d.ts

```diff
@@ -107,44 +108,59 @@ export interface Feature extends Record<string, any> {
   } | null;
 }
 
-export type Facet = RangeFacet | OptionFacet;
+export type Facet = RangeFacet | OptionFacet | HierarchialOptionFacet;
+
+export type BrowseFacet = Pick<Facet, keyof BaseFacet | 'type'>;
 
 export interface BaseFacet extends Record<string, any> {
   data: Record<string, any>;
-  status: Record<string, any>;
   display_name: string;
   name: string;
   hidden: boolean;
 }
 
 export interface RangeFacet extends BaseFacet, Record<string, any> {
-  max: number;
-  min: number;
+  status?: { min: RangeMin, max: RangeMax };
+  max: RangeMax;
+  min: RangeMin;
   type: 'range';
 }
 
 export interface OptionFacet extends BaseFacet, Record<string, any> {
   options: FacetOption[];
-  type: 'multiple' | 'single' | 'hierarchical';
+  type: 'multiple' | 'single';
+}
+
+export interface HierarchialOptionFacet extends BaseFacet, Record<string, any> {
+  options: HierarchialFacetOption[];
+  type: 'hierarchical';
 }
 
 export interface FacetOption extends Record<string, any> {
   count: number;
   display_name: string;
   value: string;
-  options?: FacetOption[];
-  range?: ['-inf' | number, 'inf' | number];
+  range?: [RangeMin, RangeMax];
   status: string;
+  data: Record<string, any>;
 }
 
+export interface HierarchialFacetOption extends FacetOption {
+  options?: HierarchialFacetOption[];
```

Comment:
Can the hierarchial facet option be recursive to to any depth? This typing assumes so. If it only allows two levels deep options, the type here should be replaced with FacetOption.

Also did not test this option so I don't know if it is always present or optional, so left it optional.